src/jdk.javadoc/share/classes/jdk/javadoc/internal/tool/Start.java

```diff
@@ -657,11 +659,34 @@ int consumeDocletOption(int idx, List<String> args, boolean isToolOption) throws
         // check if arg is accepted by the tool before emitting error
         if (!isToolOption) {
             text = log.getText("main.invalid_flag", arg);
-            throw new OptionException(ERROR, this::showUsage, text);
+            throw new OptionException(ERROR, () -> reportBadOption(arg), text);
         }
         return m * idx;
     }
 
+    private void reportBadOption(String name) {
+        var allOptionNames = Stream.concat(
+                getToolOptions().getSupportedOptions().stream()
+                        .flatMap(o -> o.getNames().stream()),
+                docletOptions.stream()
+                        .flatMap(o -> o.getNames().stream()));
+        // The following code is from the snippet in the comment for StringUtils.DamerauLevenshteinDistance
+        record Pair(String word, int distance) { }
+        var suggestions = allOptionNames
+                             .map(v -> new Pair(v, StringUtils.DamerauLevenshteinDistance.of(v, name)))
+                             .filter(p -> Double.compare(1.0 / 3, ((double) p.distance()) / p.word().length()) >= 0)
+                             .sorted(Comparator.comparingDouble(Pair::distance))
+                             .limit(3)
+                             .toList();
```

Comment:
Looking at this copied snippet, I realise that it might not be the best example I could come up with when introduced that functionality. Have you seen the actual use here? https://github.com/openjdk/jdk/blob/3aefd1ce714ad2d51f31b73c73b423c7bb7443ed/src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclint/DocLint.java#L380-L401 It might work better as it does not limit the list of findings to top 3, a number which (as we discussed offline) has nothing to do with the distance threshold of `1.0 / 3`.

Reply:
Thanks; I'll use that code.   

FWIW, I do think it is worth adding a method in StringUtils to encapsulate that behavior, if we can settle on what the functionality should be.